pythonforandroid/bootstraps/common/build/src/main/java/org/kivy/android/PythonService.java

```diff
@@ -116,7 +122,31 @@ protected void doStartForeground(Bundle extras) {
             builder.setSmallIcon(context.getApplicationInfo().icon);
             notification = builder.build();
         }
-        startForeground(1, notification);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O){
+            Log.v("python service","android 8: start foregrund with new channel");
+            startForegroundWithNewChannel(1, notification);
+        }else{
+            Log.v("python service","android <8: start foregrund withut new channel");
+            startForeground(1, notification);
+        }
+    }
+
+    /**
+     * https://stackoverflow.com/questions/47531742/startforeground-fail-after-upgrade-to-android-8-1
+     *
+     * this change is necessary because from Android 8 on we need our own channel
+     */
```

Comment:
ditto, it's a change now, but i'll just be part of the code later, which will make the comment weird, just remove "change" from the sentence.

Reply:
Resolving as "addressed" thanks!